### PR TITLE
Fixed Carthage support.

### LINKED ISSUE #585: Carthage building error on SVProgressHUD "2.0.1"
when I run "carthage update --platform iOS"
I got this error messages
______
The following build commands failed:
	Ld /Users/fortinet/Library/Developer/Xcode/DerivedData/SVProgressHUD-gvktundsqvzqwobklivjdudvqtgq/Build/Intermediates/SVProgressHUD.build/Release-iphoneos/SVProgressHUD-Framework.build/Objects-normal/armv7/SVProgressHUD normal armv7
(1 failure)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
A shell task failed with exit code 65:
** BUILD FAILED **

The following build commands failed:
	Ld /Users/fortinet/Library/Developer/Xcode/DerivedData/SVProgressHUD-gvktundsqvzqwobklivjdudvqtgq/Build/Intermediates/SVProgressHUD.build/Release-iphoneos/SVProgressHUD-Framework.build/Objects-normal/armv7/SVProgressHUD normal armv7
(1 failure)
_______

Note: svprogresshud "2.0" doesn't have this issue

### PR DESCRIPTION
**Pull request by justin**
Fixes #585

Cleaned up a few different settings in the Framework target for Carthage. I pointed our project at my branch and it builds without issue. 

Bonus points if you want to run the commands below and add the zip file output to the git tags. Then Carthage can just download that zip file and use the binaries inside rather than having to burn CPU cycles rebuilding the project from scratch.

`carthage build --no-skip-current`
`carthage archive`

### CODE REVIEW

File: SVProgressHUD.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -288,6 +288,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.transitapp.$(PRODUCT_NAME:rfc1034identifier)";
```
Comment (honkmaster) [inline SVProgressHUD.xcodeproj/project.pbxproj]: Where does `com.transitapp` come from? I thought we deleted all references some time ago?

File: SVProgressHUD.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -329,6 +329,7 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 8.3;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				MTL_ENABLE_DEBUG_INFO = NO;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.transitapp.$(PRODUCT_NAME:rfc1034identifier)";
```
Comment (honkmaster) [inline SVProgressHUD.xcodeproj/project.pbxproj]: Same as above: Where does `com.transitapp` come from? I thought we deleted all references some time ago?

**Comment by honkmaster** [comment]
Looks good to me, apart from my two questions.

File: SVProgressHUD.xcodeproj/project.pbxproj (hunk 2 above)
Comment (cartermike) [inline SVProgressHUD.xcodeproj/project.pbxproj]: com.transitapp got added in https://github.com/SVProgressHUD/SVProgressHUD/pull/409/commits/22cce213e9be952b556f68350b3087a3b6a98296

**Comment by justin** [comment]
What do you want instead of `com.transitapp.SVProgressHUD`?

**Comment by honkmaster** [comment]
Why not $(Product_bundle_identifier) as in the info.plist?

File: SVProgressHUD.xcodeproj/project.pbxproj (hunk 2 above)
Comment (honkmaster) [inline SVProgressHUD.xcodeproj/project.pbxproj]: At the time of this 22cce21 com.transitapp was valid. However as we moved away from the transitapp group to an own group this should not be mentioned any more.

**Comment by justin** [comment]
Xcode adds those settings when you validate the project. Them not being set before is part of why I think Carthage was failing on build.

**Comment by justin** [comment]
moo

**Comment by honkmaster** [comment]
But where does these values come from, as they have not been set anywhere else?

**Comment by justin** [comment]
Magic? I have no idea.

- j

**Comment by honkmaster** [comment]
could you run `grep -r "transit" .` in the project directory? It must be somewhere ... Spooky.

**Comment by justin** [comment]
OK, I see what it's doing. Xcode wants to move the value out of the Info.plist and move it into the pbxproj. I reverted that commit, so it's going through Info.plist again.

**Comment by honkmaster** [comment]
Please test the master if everything works as expected because I merged two PR regarding Carthage. If everything works i will release 2.0.3.

**Comment by justin** [comment]
I just built it without error. Thanks!

**Comment by mdcarter1** [comment]
Works fine for me now as well.

**Comment by ReedD** [comment]
Works for me too :+1:

**Pull request merged**